core/src/main/java/org/web3j/utils/Account.java

```diff
@@ -0,0 +1,55 @@
+package org.web3j.utils;
+
+import java.util.regex.Pattern;
+
+import org.web3j.crypto.Hash;
+
+/**
+ * Account utility functions.
+ */
+public class Account {
+
+    private static final Pattern ignoreCaseAddrPattern = Pattern.compile("(?i)^(0x)?[0-9a-f]{40}$");
+    private static final Pattern lowerCaseAddrPattern = Pattern.compile("^(0x)?[0-9a-f]{40}$");
+    private static final Pattern upperCaseAddrPattern = Pattern.compile("^(0x)?[0-9A-F]{40}$");
+
+    /**
+     * Verify that a hex account string is a valid Ethereum address.
+     *
+     * @param address given address in HEX
+     * @return is this a valid address
+     */
+    public static Boolean isAddress(String address) {
+        /*
+         * check basic address requirements, i.e. is not empty and contains
+         * the valid number and type of characters
+         */
+        if (address.isEmpty() || !ignoreCaseAddrPattern.matcher(address).find()) {
+            return false;
+        } else if (lowerCaseAddrPattern.matcher(address).find()
+                || upperCaseAddrPattern.matcher(address).find()) {
+            // if it's all small caps or caps return true
+            return true;
+        } else {
+            // if it is mixed caps it is a checksum address and needs to be validated
+            return validateChecksumAddress(address);
+        }
+    }
+
+    private static Boolean validateChecksumAddress(String address) {
```

Comment:
Validating checksum addresses is useful functionality.

It might be better to have isChecksumAddress as a method on the Address type, and a toChecksumAddress that operates on an Address or String instance